docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -11,410 +11,616 @@ The `ERC20.cairo` contract implements an approximation of https://eips.ethereum.
 * <<extensibility,Extensibility>>
 * <<presets,Presets>>
  ** <<erc20_basic,ERC20 (basic)>>
- ** <<erc20mintable,ERC20Mintable>>
- ** <<erc20pausable,ERC20Pausable>>
- ** <<erc20upgradeable,ERC20Upgradeable>>
 * <<api_specification,API Specification>>
- ** <<methods,Methods>>
+ ** <<erc20impl,ERC20Impl>>
   *** <<name,`name`>>
   *** <<symbol,`symbol`>>
   *** <<decimals,`decimals`>>
-  *** <<totalsupply,`totalSupply`>>
-  *** <<balanceof,`balanceOf`>>
+  *** <<total_supply,`total_supply`>>
+  *** <<balance_of,`balance_of`>>
   *** <<allowance,`allowance`>>
   *** <<transfer,`transfer`>>
-  *** <<transferfrom,`transferFrom`>>
+  *** <<transfer_from,`transfer_from`>>
   *** <<approve,`approve`>>
  ** <<events,Events>>
-  *** <<transfer_event,`Transfer (event)`>>
-  *** <<approval_event,`Approval (event)`>>
+  *** <<Transfer,`Transfer`>>
+  *** <<Approval,`Approval`>>
+ ** <<non-standard-external,Non-standard external>>
+  *** <<increase_allowance,`increase_allowance`>>
+  *** <<decrease_allowance,`decrease_allowance`>>
+ ** <<internalimpl,InternalImpl>>
+  *** <<initializer,initializer>>
+  *** <<internal_increase_allowance,_increase_allowance>>
+  *** <<internal_decrease_allowance,_decrease_allowance>>
+  *** <<internal_mint,_mint>>
+  *** <<internal_burn,_burn>>
+  *** <<internal_approve,_approve>>
+  *** <<internal_transfer,_transfer>>
+  *** <<internal_spend_allowance,_spend_allowance>>
 
 == Interface
 
-[,cairo]
-----
-@contract_interface
-namespace IERC20 {
-    func name() -> (name: felt) {
-    }
-
-    func symbol() -> (symbol: felt) {
-    }
-
-    func decimals() -> (decimals: felt) {
-    }
-
-    func totalSupply() -> (totalSupply: Uint256) {
-    }
-
-    func balanceOf(account: felt) -> (balance: Uint256) {
-    }
-
-    func allowance(owner: felt, spender: felt) -> (remaining: Uint256) {
-    }
-
-    func transfer(recipient: felt, amount: Uint256) -> (success: felt) {
-    }
-
-    func transferFrom(sender: felt, recipient: felt, amount: Uint256) -> (success: felt) {
-    }
-
-    func approve(spender: felt, amount: Uint256) -> (success: felt) {
-    }
+[,rust]
+----
+#[starknet::interface]
+trait IERC20<TState> {
+    fn name(self: @TState) -> felt252;
+    fn symbol(self: @TState) -> felt252;
+    fn decimals(self: @TState) -> u8;
+    fn total_supply(self: @TState) -> u256;
+    fn balance_of(self: @TState, account: ContractAddress) -> u256;
+    fn allowance(self: @TState, owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn transfer(ref self: TState, recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(
+        ref self: TState, sender: ContractAddress, recipient: ContractAddress, amount: u256
+    ) -> bool;
+    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
```

Comment:
I think this may be easier to read (in the UI)

```suggestion
#[starknet::interface]
trait IERC20<TState> {
    fn name(self: @TState) -> felt252;
    fn symbol(self: @TState) -> felt252;
    fn decimals(self: @TState) -> u8;
    fn total_supply(self: @TState) -> u256;
    fn balance_of(self: @TState, account: ContractAddress) -> u256;
    fn allowance(
        self: @TState,
        owner: ContractAddress,
        spender: ContractAddress
    ) -> u256;
    fn transfer(
        ref self: TState, 
        recipient: ContractAddress, 
        amount: u256
    ) -> bool;
    fn transfer_from(
        ref self: TState, 
        sender: ContractAddress, 
        recipient: ContractAddress, 
        amount: u256
    ) -> bool;
    fn approve(ref self: TState, spender: ContractAddress, amount: u256) -> bool;
```